### PR TITLE
Use unique background/analysis/ensemble/control streams

### LINKED ISSUE #23: Enable new stream types defined in MPAS-JEDI
See https://github.com/JCSDA-internal/mpas-jedi/pull/647.  Need to utilize `background`, `analysis`, `ensemble`, and `control` streams, where appropriate.

### LINKED ISSUE #25: Allow for experiment-specific configuration of inner iterations and outer iterations within config/experiment.csh

### PR DESCRIPTION
**Pull request by jjguerrette**
### Description
Use unique streams for  reading and writing `background`, `analysis`, `ensemble`, `control` (for ABE Inflation), and `da_state` (forecast output) states.  Corresponds to changes in https://github.com/JCSDA-internal/mpas-jedi/pull/647.

Closes #23 

Also adds a top-level configuration element in `config/experiment.csh` named `nInnerIterations`, which enables users to list the number of inner iterations in each outer iteration.  The yaml files for the CyclingDA task will automatically be updated with the correct number of inner and outer iterations.

Closes #25 

### Dependencies
Uses code changes from https://github.com/JCSDA-internal/mpas-jedi/pull/647.  MPAS-Workflow utilizes a static build, which is updated in this PR.  For the weekly cycling tests to pass though, this PR must be merged at the same time as the corresponding mpas-jedi PR.

### CODE REVIEW

File: config/filestructure.csh [hunk 1]
```diff
@@ -27,8 +27,9 @@ set TopExpDir = /glade/scratch/${ExperimentUser}/pandac
 setenv PackageBaseName MPAS-Workflow
 setenv ExperimentName ${ExperimentUser}
 setenv ExperimentName ${ExperimentName}_${DAType}
-setenv ExperimentName ${ExperimentName}${ExpObsName}
-setenv ExperimentName ${ExperimentName}${EnsExpSuffix}
+setenv ExperimentName ${ExperimentName}${ExpIterSuffix}
```
Comment (jjguerrette) [inline config/filestructure.csh:30]: It should be noted that the default experiment name will change due to the appending of `${ExpIterSuffix}`, which lists the number of inner iterations in each outer iteration.

Single outer, 60 inner: ExpIterSuffix=="-60-iter"
Two outer, 30 inner each: ExpIterSuffix=="-30-30-iter"

**Review by ibanos90 — APPROVED** [review]
Tested using one outer loop with 60 iterations and two outer loops with 30 iterations each one, and all changes are working fine.

**Comment by junmeiban** [comment]
Hi @jjguerrette ,
I checkout code today and would like to run 3denvar experiments and got read_fields errors (see: /glade/scratch/jban/pandac/testpr_20211018_ObsBiasWrite/DA/2018041500/da.log).
Your PR( https://github.com/JCSDA-internal/mpas-jedi/pull/647/files)
mentioned here are related examples for how to use this new function, but I do not know which commits are related to the new function.
Would you please point it to me and I can correct my yaml and namelist. 
Many thanks,
Junmei

**Comment by jjguerrette** [comment]
Hi @junmeiban,
It looks like you need to modify `streams.atmosphere` in `/glade/scratch/jban/pandac/testpr_20211018_ObsBiasWrite/DA/2018041500/` so that it includes `background`, `analysis`, and `ensemble` streams.  There is an example that was introduced in this MPAS-Workflow PR here:

https://github.com/NCAR/MPAS-Workflow/blob/develop/config/mpas/variational/streams.atmosphere

-JJ

**Comment by junmeiban** [comment]
Thanks @jjguerrette. It works now according to your suggestions.

**Pull request merged**